ShowerShow/Control/UserPreferencesController.cs

```diff
@@ -0,0 +1,119 @@
+            }
+            else
+            {
+                await _userPrefencesService.CreateUserPreferences(createPreferencesDTO);
+                HttpResponseData responseData = req.CreateResponse();
+                responseData.StatusCode = HttpStatusCode.Created;
+
+                return responseData;
+            }
+        }
+        [Function("GetUserPreferenceById")]
+        [OpenApiOperation(operationId: "GetAllPreferences", tags: new[] { "Preferences" })]
+        [OpenApiParameter(name: "userId", In = ParameterLocation.Path, Required = true, Type = typeof(Guid), Description = "The user ID parameter")]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(List<Preferences>), Description = "Successfully retrieved the Preferences")]
+        public async Task<HttpResponseData> GetUserPreferenceById([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "user/{userId:Guid}/preferences")] HttpRequestData req, Guid userId)
+        {
+            _logger.LogInformation($"Fetching the user preference by id {userId}");
+
+            if (await _userService.CheckIfUserExistAndActive(userId))
+            {
+                HttpResponseData responseData = req.CreateResponse();
+                responseData.StatusCode = HttpStatusCode.BadRequest;
+                return responseData;
+            }
+            else
+            {
+                var preferences = await _userPrefencesService.GetUserPreferencesById(userId);
+                HttpResponseData responseData = req.CreateResponse();
+                await responseData.WriteAsJsonAsync(preferences);
+                responseData.StatusCode = HttpStatusCode.OK;
+
+                return responseData;
+            }
+        }
+        [Function("UpdatePreferenceById")]
+        [OpenApiOperation(operationId: "UpdatePreferences", tags: new[] { "Preferences" })]
+        [OpenApiParameter(name: "userId", In = ParameterLocation.Path, Required = true, Type = typeof(Guid), Description = "Id of the requested user ")]
+        [OpenApiRequestBody("application/json", typeof(UpdatePreferencesDTO), Description = "The user data to update.")]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(UpdatePreferencesDTO), Description = "The OK response with the updated user preference")]
+        public async Task<HttpResponseData> UpdatePreferenceById([HttpTrigger(AuthorizationLevel.Anonymous, "put", Route = "user/{userId:Guid}/preferences")] HttpRequestData req, Guid userId)
+        {
+            _logger.LogInformation($"Fetching the user by id {userId}");
+
+            string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+            UpdatePreferencesDTO updatePreferencesDTO = JsonConvert.DeserializeObject<UpdatePreferencesDTO>(requestBody);
+
+            if (await _userService.CheckIfUserExistAndActive(userId))
```

Comment:
You are checking if the user is active and then returning bad request.. Shouldnt you do the opposite?